src/OpenTelemetry.Exporter.Jaeger/Implementation/JaegerUdpBatcher.cs

```diff
@@ -72,13 +70,23 @@ public JaegerUdpBatcher(JaegerExporterOptions options, TTransport clientTranspor
 
             this.maxFlushIntervalTimer.Elapsed += async (sender, args) =>
             {
+                bool lockTaken = this.flushLock.Wait(0);

```

Comment:
This is an optimization I added in BatchingActivityProcessor & JaegerUdpBatcher. If the timer fires, and the flush lock is already held, it will just return, releasing the thread instead of blocking. The idea being, when the running flush finishes it will restart the timer anyway. This should eliminate a bunch of threads stacking up waiting on the lock.